File: vol1/ch5/LeetCode205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        hmap1,hmap2={},{}
        for i in range(0,len(s)):
            x,y=s[i],t[i]
            if x not in hmap1 and y not in hmap2:
                hmap1[x],hmap2[y]=y,x
            elif (x in hmap1 and hmap1[x]!=y) or (y in hmap2 and hmap2[y]!=x):
                return False					#不匹配返回false
        return True

File: vol1/ch5/LeetCode205.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if (s.size()!=t.size())
            return false;
        unordered_map<char,int> hmap1;
        unordered_map<char,int> hmap2;
        for (int i=0;i<s.size();i++) {
            char x=s[i],y=t[i];
            if(hmap1.count(x)==0 && hmap2.count(y)==0) {
                hmap1[x]=y;
                hmap2[y]=x;
            }
            else if (hmap1.count(x)==1 && hmap1[x]!=y || hmap2.count(y)==1 && hmap2[y]!=x)
                return false;								//²»Æ¥Åä·µ»Øfalse
        }
        return true;
    }
};

